src/lib/file-uploader.js

```diff
@@ -2,6 +2,31 @@ import {BitmapAdapter} from 'scratch-svg-renderer';
 import randomizeSpritePosition from './randomize-sprite-position.js';
 import bmpConverter from './bmp-converter';
 import gifDecoder from './gif-decoder';
+import DOMPurify from 'dompurify';
+
+let _TextDecoder;
+let _TextEncoder;
+if (typeof TextDecoder === 'undefined' || typeof TextEncoder === 'undefined') {
+    // Wait to require the text encoding polyfill until we know it's needed.
+    // eslint-disable-next-line global-require
+    const encoding = require('fastestsmallesttextencoderdecoder');
```

Comment:
This needs to be added to package.json. (That said, the only supported browser which requires this code path is Old Edge - which is discontinued. Therefore it might make sense to just use the global.) GUI currently uses `text-encoding`, but I believe that library is unnecessarily big and should be replaced with this one.

Reply:
Thanks for pointing this out, @apple502j. We're still using `fastestsmallesttextencoderdecoder` but we've moved the bulk of this code to scratch-svg-renderer https://github.com/LLK/scratch-svg-renderer/pull/336